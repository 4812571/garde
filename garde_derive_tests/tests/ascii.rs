#[path = "./util/mod.rs"]
mod util;

#[derive(Debug, garde::Validate)]
struct Test<'a> {
    #[garde(ascii)]
    field: &'a str,
}

#[test]
fn ascii_valid() {
    util::check_ok(&[Test { field: "a!0_~" }], &())
}

#[test]
fn ascii_invalid() {
    util::check_fail!(&[Test { field: "ğŸ˜‚" }], &())
}
