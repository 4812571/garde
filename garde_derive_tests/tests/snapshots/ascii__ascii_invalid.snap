---
source: garde_derive_tests/tests/ascii.rs
expression: snapshot
---
Test {
    field: "ğŸ˜‚",
}
`field` is not ascii


